ui/admin/app/templates/scopes/scope/roles/role/scopes.hbs

```diff
@@ -21,137 +21,192 @@
+                    {{t
+                      (concat
+                        'resources.role.scope.keywords.' B.data.id '.sub-title'
+                      )
+                    }}
+                  </Hds::Text::Body>
                 {{else}}
-                  {{B.data.displayName}}
-                {{/if}}
-              {{/if}}
-            </B.Td>
-            <B.Td>
-              <Hds::Copy::Snippet
-                @textToCopy={{B.data.id}}
-                @color='secondary'
-              />
-            </B.Td>
-            <B.Td>
-              {{#if B.data.scope.isGlobal}}
-                <Hds::Link::Inline
-                  @route='scopes.scope'
-                  @model={{B.data.scope.id}}
-                  @icon='globe'
-                  @iconPosition='leading'
-                >
-                  {{#if B.data.scope.name}}
-                    {{t 'titles.global'}}
+                  {{#if (can 'read model' B.data)}}
+                    <LinkTo @route='scopes.scope' @model={{B.data}}>
+                      {{#if B.data.isGlobal}}
+                        {{t 'titles.global'}}
+                      {{else}}
+                        {{B.data.displayName}}
+                      {{/if}}
+                    </LinkTo>
                   {{else}}
-                    {{B.data.scope.id}}
+                    {{B.data.displayName}}
                   {{/if}}
-                </Hds::Link::Inline>
-              {{else if B.data.scope.isOrg}}
-                <Hds::Link::Inline
-                  @route='scopes.scope'
-                  @model={{B.data.scope.id}}
-                  @icon='org'
-                  @iconPosition='leading'
-                >
-                  {{#if B.data.scope.name}}
-                    {{B.data.scope.name}}
-                  {{else}}
-                    {{B.data.scope.id}}
-                  {{/if}}
-                </Hds::Link::Inline>
-              {{else}}
-                {{#if B.data.scope.name}}
-                  {{B.data.scope.name}}
-                {{else}}
-                  {{B.data.scope.id}}
                 {{/if}}
-              {{/if}}
-            </B.Td>
-            <B.Td>
-              {{#if B.data.type}}
-                <Hds::Badge
-                  @text={{t (concat 'resources.scope.types.' B.data.type)}}
+              </B.Td>
+              <B.Td>
+                <Hds::Copy::Snippet
+                  @textToCopy={{B.data.id}}
+                  @color='secondary'
                 />
-              {{/if}}
-            </B.Td>
-          </B.Tr>
-        </:body>
-      </Hds::Table>
+              </B.Td>
+              <B.Td>
+                {{#if B.data.scope.isGlobal}}
+                  <Hds::Link::Inline
+                    @route='scopes.scope'
+                    @model={{B.data.scope.id}}
+                    @icon='globe'
+                    @iconPosition='leading'
+                  >
+                    {{~if
```

Comment:
the `~` inside the curly braces is used to control whitespace in hbs. It removed the extra space caused by the newline.

Reply:
TIL

Reply:
Is there CSS or something else that is making the whitespace in the HTML not collapse?

Reply:
hmm good point, im not sure, ill investigate

Reply:
looked into it and it seems to be an ember [hbs bug](https://github.com/emberjs/ember.js/issues/19603#issuecomment-1121406604) with rendering extra whitespace. And there wasn't any css (that i could find) that would cause this whitespace issue.

Reply:
Wow interesting, good find! Thanks for the info